src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -24,7 +24,7 @@
 namespace System
 {
     [Serializable]
-    [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
+    [TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
     public abstract partial class Enum : ValueType, IComparable, IFormattable, IConvertible
```

Comment:
I'm sure this is deliberate, but why not implement **ISpanFormattable**?

Reply:
Every call to TryFormat would box the enum, just as every call to ToString() today boxes the enum.  To avoid that would require non-trivial work in the JIT to rewrite calls from the instance method to something else.  Unlike GetHashCode and HasFlags, which are special-cased, the implementation of formatting is complicated and not something we'd want to replicate in the JIT itself, so most likely we'd need a static generic TryFormat anyway the JIT could rewrite the calls to target.  And we don't want to expose an ISpanFormattable implementation that's going to implicitly have such expense.

Reply:
@stephentoub thanks for taking the time to respond, I really appreciate it! Can't you use constrained calls with enums to avoid boxing (like what's done [here](https://github.com/dotnet/runtime/blob/acba5872a6437f9dc622319d88042e113a49c81c/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/DefaultInterpolatedStringHandler.cs#L359), for example)?

PS. Last question, I promise :p

Reply:
No.  Try just running:
```C#
for (int i = 0; i < 1000; i++) DayOfWeek.Monday.ToString();
```
under an allocation profiler.  You'll see it allocates 1000 enum objects.  The implementation of ToString itself is defined on the System.Enum reference type.